packages/shared/src/merge.ts

```diff
@@ -79,13 +79,17 @@ function propertyIsOnObject(object: any, property: any) {
 
 // Protects from prototype poisoning and unexpected merging up the prototype chain.
 function propertyIsUnsafe(target, key) {
-  return (
-    propertyIsOnObject(target, key) && // Properties are safe to merge if they don't exist in the target yet,
-    !(
-      Object.hasOwnProperty.call(target, key) && // unsafe if they exist up the prototype chain,
-      Object.propertyIsEnumerable.call(target, key)
-    )
-  ) // and also unsafe if they're nonenumerable.
+  try {
+    return (
+      propertyIsOnObject(target, key) && // Properties are safe to merge if they don't exist in the target yet,
+      !(
+        Object.hasOwnProperty.call(target, key) && // unsafe if they exist up the prototype chain,
+        Object.propertyIsEnumerable.call(target, key)
+      )
+    ) // and also unsafe if they're nonenumerable.
+  } catch (_) {
+    return true
```

Comment:
Here will lose some cached props. But if returns `false`, there will still be something wrong between rn and immer.